jwst/associations/lib/rules_level3.py

```diff
@@ -667,6 +667,25 @@ def __init__(self, *args, **kwargs):
                 ],
                 reduce=Constraint.notany
             ),
+            # Don't allow NIRCam engineering mode
+            # with PUPIL='CLEAR' in tso3
+            Constraint(
+                [
+                    Constraint([
+                        DMSAttrConstraint(
+                            name='restricted_grism',
+                            sources=['exp_type'],
+                            value = ('nrc_tsgrism')
+                        ),
+                        DMSAttrConstraint(
+                            name='grism_clear',
```

Comment:
Same thing here, where I didn't want to reuse the name `pupil`, so I used `grism_clear` instead.